### PR TITLE
auth and 404 page

### PR DESCRIPTION
**Pull request by Chiragmodi01**
- Auth
  - signup, login and logout
 - 404 page

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import './App.css';
-import React, {useRef} from 'react';
+import React, {useEffect, useRef} from 'react';
 import {FooterNav, Navbar, Navpills, Sidebar} from './comps/index'
-import {Home, Playlist, Liked, Watchlater, History, Signup, Login} from './pages/index';
+import {Home, Playlist, Liked, Watchlater, History, Signup, Login, PageNotFound} from './pages/index';
```
Comment (ravikumar0403) [inline src/App.js:4]: `/index` is not required as it will check the index files for exports if you do not mention any file

File: src/App.js [hunk 2]
```diff
@@ -17,25 +19,35 @@ function App() {
...
+          <Route path="/" element={<Home />}/>
+          <Route path="/playlist" element={<Playlist />}/>
+          <Route path="/liked" element={<Liked />}/>
+          <Route path="/watchlater" element={<Watchlater />}/>
+          <Route path="/history" element={<History />}/>
+          <Route path="/login" element={<Login />}/>
+          <Route path="/signup" element={<Signup />}/>
+          <Route path="/mock" element={<Mockman />} />
+          <Route path = '*' element={<PageNotFound />} />
+          {/* <Route path="/settings" element={<Settings />}/> */}
```
Comment (ravikumar0403) [inline src/App.js:47]: remove commented code if not required

File: src/assets/gif/index.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import pagenotfound from './pagenotfound.gif';
+
+export {pagenotfound};
```
Comment (ravikumar0403) [inline src/assets/gif/index.js:3]: you don't have to import and then export, you can write like this instead
`export default as { componentName }  from "path/to/file"` if you have exported as default from the component file

File: src/comps/Sidebar/Sidebar.jsx [hunk 4]
```diff
@@ -3,36 +3,43 @@ import './Sidebar.css'
 
 import {AiFillHome, AiOutlineHome, AiFillClockCircle, AiOutlineClockCircle, AiOutlineHistory, RiThumbUpFill, RiThumbUpLine, MdVideoLibrary, MdOutlineVideoLibrary} from '../../utils/getIcons';
 import { useMain } from '../../helpers/context/main-context';
-import { NavLink, useLocation } from "react-router-dom";
+import { NavLink, useLocation, useNavigate } from "react-router-dom";
 
 function Sidebar() {
```
Comment (ravikumar0403) [inline src/comps/Sidebar/Sidebar.jsx:8]: why not es6 arrow functions?

**Review by ravikumar0403 — APPROVED** [review]

File: src/comps/Navbar/Navbar.jsx [hunk 5]
```diff
@@ -1,12 +1,14 @@
 import React  from 'react'
 import './Navbar.css'
 import {useMain} from '../../helpers/context/main-context';
-import {IoIosMenu, IoSearchOutline, CgProfile, BiDotsVerticalRounded, BsMoon, BsSun, IoIosArrowForward, IoSettingsOutline} from '../../utils/getIcons';
+import {IoIosMenu, IoSearchOutline, CgProfile, BiDotsVerticalRounded, BsMoon, BsSun, IoIosArrowForward, IoSettingsOutline, AiOutlineLogout} from '../../utils/getIcons';
 import {StyledButton, StyledDropdown} from '../../styledComps/index';
 import { Link } from "react-router-dom";
+import verifyForm from '../../utils/verifyForm';
 
 function Navbar({toggleDropdownRef}) {
```
Comment (aurobindoGupta) [inline src/comps/Navbar/Navbar.jsx:9]: come on bhai show us some of ur ES6 skills.!!

**Review by aurobindoGupta — APPROVED** [review]
keep up the good work, try to start coding in es6.

File: src/assets/gif/index.js (hunk 3 above)
Comment (Chiragmodi01) [reply src/assets/gif/index.js:3]: images are not exported by themselves in the same file, so I've to import them and then export.

File: src/comps/Navbar/Navbar.jsx (hunk 5 above)
Comment (Chiragmodi01) [reply src/comps/Navbar/Navbar.jsx:9]: old time preferences

File: src/comps/Sidebar/Sidebar.jsx (hunk 4 above)
Comment (Chiragmodi01) [reply src/comps/Sidebar/Sidebar.jsx:8]: old time preferences

**Pull request merged**